zerver/lib/bugdown/__init__.py

```diff
@@ -545,6 +545,18 @@ def youtube_image(self, url: Text) -> Optional[Text]:
             return "https://i.ytimg.com/vi/%s/default.jpg" % (yt_id,)
         return None
 
+    def get_youtube_title(self, url: Text) -> Optional[Text]:
+        request_url = "https://noembed.com/embed?url=" + url
```

Comment:
I think you should not rely on a third party to get this info. Doesn't YouTube provide an official API to do this?